Cql/Cql.Compiler/LibraryExpressionBuilderContext.cs

```diff
@@ -13,27 +13,40 @@
 
 namespace Hl7.Cql.Compiler;
 
-internal partial class LibraryExpressionBuilderContext(
-    ILogger<LibraryExpressionBuilder> logger,
-    ExpressionBuilder expressionBuilder,
-    Library library,
-    DefinitionDictionary<LambdaExpression> libraryDefinitions,
-    LibrarySetExpressionBuilderContext? libsCtx = null)
+internal partial class LibraryExpressionBuilderContext
 {
-    private readonly ILogger<LibraryExpressionBuilder> _logger = logger;
-    private readonly ExpressionBuilder _expressionBuilder = expressionBuilder;
+    private readonly ILogger<LibraryExpressionBuilder> _logger;
+    private readonly ExpressionBuilder _expressionBuilder;
+    private readonly DefinitionDictionary<LambdaExpression> _libraryDefinitions;
+    private readonly LibrarySetExpressionBuilderContext? _libsCtx;
+
+    public LibraryExpressionBuilderContext(
+        ILogger<LibraryExpressionBuilder> logger,
+        ExpressionBuilder expressionBuilder,
+        Library library,
+        DefinitionDictionary<LambdaExpression> libraryDefinitions,
+        LibrarySetExpressionBuilderContext? libsCtx = null)
+    {
+        _libraryDefinitions = libraryDefinitions;
+        _libsCtx = libsCtx;
+        _logger = logger;
+        _expressionBuilder = expressionBuilder;
+        Library = library;
+        LibraryVersionedIdentifier = Library.GetVersionedIdentifier()!;
+    }
+
     private static readonly AmbiguousOverloadCorrector AmbiguousOverloadCorrector = new AmbiguousOverloadCorrector();
 
     /// <summary>
     /// Gets the library associated with the expression builder context.
     /// </summary>
-    public Library Library => library;
+    public Library Library { get; }
```

Comment:
Re-using the class constructor parameters is exactly the pattern microsoft wants us to use, why are you taking the long route here?

Reply:
This is public property is mapping to a primary constructor parameter. This is a class, which is not the same as a record when it also becomes the property